kong/tools/public.lua

```diff
@@ -12,62 +12,139 @@ do
 
 
   local str_find              = string.find
+  local str_format            = string.format
   local ngx_req_get_post_args = ngx.req.get_post_args
   local ngx_req_get_body_data = ngx.req.get_body_data
 
 
-  local content_type_map = {
-    [1] = function(content_type) -- multipart
-      return multipart(ngx_req_get_body_data(), content_type):get_all()
+  local MIME_TYPES = {
+    form_url_encoded = 1,
+    json             = 2,
+    xml              = 3,
+    multipart        = 4,
+    text             = 5,
+    html             = 6,
+  }
+
+
+  local ERRORS     = {
+    no_ct          = 1,
+    [1]            = "don't know how to parse request body: no Content-Type",
+    unknown_ct     = 2,
+    [2]            = "don't know how to parse request body: " ..
```

Comment:
it feels odd to see multiple colons in the same fragment. perhaps this could read something like

`don't know how to parse request body (unknown Content-Type '%s')`